### PR TITLE
Create dynamic configuration

### PR DESCRIPTION
**Pull request by xhernandez**
These PR introduces a dynamically created configuration file that can easily be tailored for different operating systems and backends.

This configuration will be used in a next PR to parametrize cluster settings (number of nodes, resources, ...)

### CODE REVIEW

**Comment by anoopcs9** [comment]
First impression:

I was going through the changes commit by commit and https://github.com/samba-in-kubernetes/sit-environment/pull/8/commits/89d89f46682120eef6de2eb3a05f25a6a6bd758e seems like an overkill especially with all those **make** functions for conditionals and string substitution. Can we work towards a simplified approach?

**Comment by xhernandez** [comment]
> First impression:
> 
> I was going through the changes commit by commit and https://github.com/samba-in-kubernetes/sit-environment/pull/8/commits/89d89f46682120eef6de2eb3a05f25a6a6bd758e seems like an overkill especially with all those **make** functions for conditionals and string substitution. Can we work towards a simplified approach?

I'm a bit surprised by this comment given the complexity of other Makefiles in other SINK repos. I think this is a very simplified way to achieve the changes I wanted. The only thing I can do is to remove it.

Is the use of SIT_OS and SIT_BE ok, or should I restrict it to EXTRA_VARS?

**Comment by spuiuk** [comment]
Just want to point out that though the CI seems to have passed, it hasn't really. This is the testing result on the client.

--
+ EXTRA_VARS=test_sanity_only=1
+ make test
Makefile:10: warning: overriding recipe for target 'test'
Makefile:7: warning: ignoring old recipe for target 'test'
make: Circular test <- test dependency dropped.
--

https://jenkins-samba.apps.ocp.cloud.ci.centos.org/job/samba_gluster-integration-environment/152/consoleText

**Comment by anoopcs9** [comment]
> > I was going through the changes commit by commit and [89d89f4](https://github.com/samba-in-kubernetes/sit-environment/commit/89d89f46682120eef6de2eb3a05f25a6a6bd758e) seems like an overkill especially with all those **make** functions for conditionals and string substitution. Can we work towards a simplified approach?
> 
> I'm a bit surprised by this comment given the complexity of other Makefiles in other SINK repos. I think this is a very simplified way to achieve the changes I wanted. The only thing I can do is to remove it.

nvm, you can fix the Makefile warnings and I'll go through it again.

File: vagrant/roles/setup.prep/tasks/main.yml [hunk 1]
```diff
@@ -22,10 +22,18 @@
...
+    recurse: yes
+
 - name: copy ssh config in place
   copy:
     src: ansible/ssh-config-setup
     dest: /home/vagrant/.ssh/config
     mode: 0600
     owner: vagrant
     group: vagrant
+
```
Comment (anoopcs9) [inline vagrant/roles/setup.prep/tasks/main.yml:39]: nit: new blank line at EOF.

**Review by anoopcs9 — APPROVED** [review]
lgtm.

You can remove the extra blank line(see below) if you happen to refresh the patch to address any other reviews. Otherwise just leave it.

**Comment by xhernandez** [comment]
> lgtm.
> 
> You can remove the extra blank line(see below) if you happen to refresh the patch to address any other reviews. Otherwise just leave it.

If there are no other changes, I'll do it in next PR.

Would it make sense to fix all issues detected by yamllint ? many are for long lines (> 80 columns) and using yes/no instead of true/false. I'm not sure if we should be strict about that.

**Comment by anoopcs9** [comment]
> Would it make sense to fix all issues detected by yamllint ? many are for long lines (> 80 columns) and using yes/no instead of true/false. I'm not sure if we should be strict about that.

How about creating an issue for further discussion? We could even think of adding the `yamllint` support as we do with other SINK repositories.

**Comment by xhernandez** [comment]
> > Would it make sense to fix all issues detected by yamllint ? many are for long lines (> 80 columns) and using yes/no instead of true/false. I'm not sure if we should be strict about that.
> 
> How about creating an issue for further discussion? We could even think of adding the `yamllint` support as we do with other SINK repositories.

Done: #10

**Review by spuiuk — APPROVED** [review]
ACK

**Pull request merged**